src/code/z_skelanime.c

```diff
@@ -1,176 +1,1680 @@
+        } else {
+            if (skelDlist[1] != 0) {
+                Matrix_ToMtx(limbMatricies, "../z_skelanime.c", 1334);
+                limbMatricies++;
+            }
+        }
+    }
+
+    if (appendDlistFunc != 0) {
+        appendDlistFunc(globalCtx, 1, &skelDlist[1], &rot, actor);
+    }
+
+    if (rootLimb->firstChildIndex != LIMB_DONE) {
+        SkelAnime_DrawLimbSV(globalCtx, rootLimb->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, &limbMatricies);
+    }
+
+    Matrix_Pull();
+    func_800C6B54(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 1347);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/SkelAnime_DrawSV.s")
+#endif
+#undef NON_MATCHING
+
+/*
+ * Copies the rotation values from the rotation value table, indexed by the rotation index table
+ * When a rotation index is >= the animation limit, the output rotation value is copied from the frame's
+ * rotation value list, otherwise it is copied from the initial rotation value list
+ */
+void SkelAnime_AnimateFrame(u32 animation, s32 currentFrame, s32 limbCount, Vec3s* dst) {
+    AnimationRotationIndex* index;
+    AnimationRotationValue* rotationValueTable;
+    AnimationRotationValue* frameRotationValueTable;
+    AnimationHeader* animationHeader;
+
+    s32 i;
+    u16 limit;
+
+    animationHeader = SEGMENTED_TO_VIRTUAL(animation);
+    index = SEGMENTED_TO_VIRTUAL(animationHeader->rotationIndexSeg);
+    rotationValueTable = SEGMENTED_TO_VIRTUAL(animationHeader->rotationValueSeg);
+    frameRotationValueTable = &rotationValueTable[currentFrame];
+    limit = animationHeader->limit;
+
+    for (i = 0; i < limbCount; i++, dst++, index++) {
+        if ((dst == NULL) || (index == NULL) || (frameRotationValueTable == NULL) || (rotationValueTable == NULL)) {
+            if (1) {} // Necessary to match.
+            LogUtils_LogThreadId("../z_skelanime.c", 1392);
+            osSyncPrintf("out = %08x\n", dst);
+            LogUtils_LogThreadId("../z_skelanime.c", 1393);
+            osSyncPrintf("ref_tbl = %08x\n", index);
+            LogUtils_LogThreadId("../z_skelanime.c", 1394);
+            osSyncPrintf("frame_tbl = %08x\n", frameRotationValueTable);
+            LogUtils_LogThreadId("../z_skelanime.c", 1395);
+            osSyncPrintf("tbl = %08x\n", rotationValueTable);
+        }
+
+        dst->x = index->x >= limit ? frameRotationValueTable[index->x] : rotationValueTable[index->x];
+        dst->y = index->y >= limit ? frameRotationValueTable[index->y] : rotationValueTable[index->y];
+        dst->z = index->z >= limit ? frameRotationValueTable[index->z] : rotationValueTable[index->z];
+    }
+}
+
+s16 func_800A1FC8(u32 animationSeg) {
+    AnimationHeader* animation = (u16*)SEGMENTED_TO_VIRTUAL(animationSeg);
+    return animation->frameCount;
+}
+
+s32 SkelAnime_GetFrameCount(u32 animationSeg) {
+    AnimationHeader* animation = (u16*)SEGMENTED_TO_VIRTUAL(animationSeg);
```

Comment:
I don't think these u16* casts are necessary or even appropriate but I could be wrong.